src/main/java/org/openhab/binding/zwave/internal/protocol/ZWaveNode.java

```diff
@@ -1475,27 +1477,41 @@ public void run() {
                 logger.trace("NODE {}: WakeupTimerTask Already asleep", getNodeId());
                 return;
             }
+            count = count + 1;
+            logger.debug("NODE {}: WakeupTimerTask {} Messages waiting, state {} count {}", getNodeId(),
+                    controller.getSendQueueLength(getNodeId()), getNodeInitStage(), count);
 
-            logger.debug("NODE {}: WakeupTimerTask {} Messages waiting, state {}", getNodeId(),
-                    controller.getSendQueueLength(getNodeId()), getNodeInitStage());
-            if (triggered == false) {
-                logger.trace("NODE {}: WakeupTimerTask First iteration", getNodeId());
+            if (isInitializationComplete() == true  && controller.getSendQueueLength(getNodeId()) == 0) {
+                triggered = true;
+            }     
+            if (count == awakemax || count == 40) {
                 triggered = true;
+            }
+            if (triggered == false) {
+                if (count == 4) {
```

Comment:
Why count=4?

constants should be defined at the top of the class.

Reply:
I'm also a bit confused about what you're trying to accomplish here.

It looks like you're allowing this timer task to execute up to `awakemax` or 40 times (based on line 1487), but you're only kicking the queue on iteration #4. Why?

Or, were you intending to kick the queue on every 4th iteration of the timer task? If so, checking for `count == 4` isn't going to do that.

Reply:
First I thought I had "count" in the right place for it to be reset.  In this case I copied/patterned after "triggerred"
```
    private class WakeupTimerTask extends TimerTask {
        // Sleep message after max battery awake duration or 20 seconds
        private boolean triggered;
        private int count;
        private final ZWaveWakeUpCommandClass wakeUpCommandClass;
        private int awakeMax = (controller.getSystemMaxAwakePeriod() * 2);
```
```
           count = 0;
            triggered = false;
```
What is going on here: posssible tl;dr
My testing uncovered that if initialization does not complete the first time, the current kickQueue on the subsequent wakes (user pressing the button) is 250 ms before the device is set awake, so nothing happens, ironically until the "Go to sleep" message is queued.  That triggers the one configuration message stuck in the queue, immediately followed by the "go to sleep" message.  I have logs in the pdf document that show this.  

Initially I had this at 0.5 seconds (count=1), but since it is hopefully going to be rare after everyone cranks up the maxAwakePeriod, I moved it 2.0 seconds (count=4), so that it did not potentially disrupt "normal" business at least for me (88% of my activity is under 2 seconds). It is only needed once, and even with the default of 5 seconds will still provide good progress on initialization. On my tests, you can get 10-15 messages per second.  The total number of messages needed for initialization varies (Fibaro and Aeotec tend to require a lot), but 30-45 messages will go a long way.  Plus if I can get the doomed AddNode Stop serial message fixed, the initial configuration time at the default setting will be 5 seconds, not 1 second like it is now.

Reply:
I see. Thanks for the explanation. If @cdjackson agrees, to me this would be worthy of a (concise) comment in the code explaining what's going on.

Reply:
Added a comment in the code